### ISSUE TITLE
Vitrail should handle intermittent connection loss to Abbot

### ISSUE THREAD

**Issue by crantila** [CANTUS-Project/vitrail#60]
While working on Progressive Web App features, I realized that Vitrail doesn't deal well with intermittently losing the connection to Abbot. In particular I'm looking for Vitrail to handle the following situations (I may add more as I think about this).

1. The HATEOAS data can't be loaded at first. It's loaded later when the user tries a request to Abbot.
1. When the user tries a request to Abbot, but it fails because Abbot's not available, the user should get the same error message whether or not the HATEOAS data were loaded.

In other words, Vitrail should transition painlessly between situations where Abbot is and is not available.
